packages/paypal-commerce-integration/src/paypal-commerce-fastlane/paypal-commerce-fastlane-payment-strategy.ts

```diff
@@ -348,15 +354,19 @@ export default class PaypalCommerceFastlanePaymentStrategy implements PaymentStr
             billingAddress: this.paypalCommerceFastlaneUtils.mapBcToPayPalAddress(billingAddress),
         });
 
-        const { orderId } = await this.paypalCommerceRequestSender.createOrder(methodId, {
-            cartId,
-            fastlaneToken: id,
-        });
+        const is3DSEnabled =
+            this.isPaypalCommerceFastlaneThreeDSAvailable() && paymentMethod.config.is3dsEnabled;
 
-        const fastlaneToken =
-            this.isPaypalCommerceFastlaneThreeDSAvailable() && paymentMethod.config.is3dsEnabled
-                ? await this.get3DSNonce(id)
-                : id;
+        if (!is3DSEnabled) {
+            const { orderId } = await this.paypalCommerceRequestSender.createOrder(methodId, {
+                cartId,
+                fastlaneToken: id,
+            });
+
+            this.orderId = orderId;
```

Comment:
It might be better to create separate method and place this code there, since it used in 2 places. The only difference between 2 similar codes is in fastlaneToken value, that could be passed as a prop.